File: omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/driver/pod/definition/NakErrorType.java
package info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.pod.definition;

public enum NakErrorType {
    FLASH_WRITE((byte) 0x01),
    FLASH_ERASE((byte) 0x02),
    FLASH_OPERATION((byte) 0x03),
    FLASH_ADDR((byte) 0x04),
    POD_STATE((byte) 0x05),
    CRITICAL_VARIABLE((byte) 0x06),
    ILLEGAL_PARAM((byte) 0x07),
    BOLUS_CRITICAL_VAR((byte) 0x08),
    INT_ILLEGAL_PARAM((byte) 0x09),
    ILLEGAL_CHECKSUM((byte) 0x0a),
    INVALID_MSG_LEN((byte) 0x0b),
    PUMP_STATE((byte) 0x0c),
    ILLEGAL_COMMAND((byte) 0x0d),
    ILLEGAL_FILL_STATE((byte) 0x0e),
    MAX_READWRITE_SIZE((byte) 0x0f),
    ILLEGAL_READ_ADDRESS((byte) 0x10),
    ILLEGAL_READ_MEM_TYPE((byte) 0x11),
    INIT_POD((byte) 0x12),
    ILLEGAL_CMD_STATE((byte) 0x13),
    ILLEGAL_SECURITY_CODE((byte) 0x14),
    POD_IN_ALARM((byte) 0x15),
    COMD_NOT_SET((byte) 0x16),
    ILLEGAL_RX_SENS_VALUE((byte) 0x17),
    ILLEGAL_TX_PKT_SIZE((byte) 0x18),
    OCCL_PARAMS_ALREADY_SET((byte) 0x19),
    OCCL_PARAM((byte) 0x1a),
    ILLEGAL_CDTHR_VALUE((byte) 0x1b),
    IGNORE_COMMAND((byte) 0x1c),
    INVALID_CRC((byte) 0x1d),
    UNKNOWN((byte) 0xff);

    private byte value;

    NakErrorType(byte value) {
        this.value = value;
    }

    public static NakErrorType byValue(byte value) {
        for (NakErrorType type : values()) {
            if (type.value == value) {
                return type;
            }
        }
        return UNKNOWN;
    }
}

File: pump/omnipod/dash/src/main/kotlin/app/aaps/pump/omnipod/dash/driver/pod/definition/NakErrorType.kt
package app.aaps.pump.omnipod.dash.driver.pod.definition

import app.aaps.pump.omnipod.dash.driver.pod.util.HasValue

enum class NakErrorType(override val value: Byte) : HasValue {

    FLASH_WRITE(0x01.toByte()),
    FLASH_ERASE(0x02.toByte()),
    FLASH_OPERATION(0x03.toByte()),
    FLASH_ADDR(0x04.toByte()),
    POD_STATE(0x05.toByte()),
    CRITICAL_VARIABLE(0x06.toByte()),
    ILLEGAL_PARAM(0x07.toByte()),
    BOLUS_CRITICAL_VAR(0x08.toByte()),
    INT_ILLEGAL_PARAM(0x09.toByte()),
    ILLEGAL_CHECKSUM(0x0a.toByte()),
    INVALID_MSG_LEN(0x0b.toByte()),
    PUMP_STATE(0x0c.toByte()),
    ILLEGAL_COMMAND(0x0d.toByte()),
    ILLEGAL_FILL_STATE(0x0e.toByte()),
    MAX_READWRITE_SIZE(0x0f.toByte()),
    ILLEGAL_READ_ADDRESS(0x10.toByte()),
    ILLEGAL_READ_MEM_TYPE(0x11.toByte()),
    INIT_POD(0x12.toByte()),
    ILLEGAL_CMD_STATE(0x13.toByte()),
    ILLEGAL_SECURITY_CODE(0x14.toByte()),
    POD_IN_ALARM(0x15.toByte()),
    COMD_NOT_SET(0x16.toByte()),
    ILLEGAL_RX_SENS_VALUE(0x17.toByte()),
    ILLEGAL_TX_PKT_SIZE(0x18.toByte()),
    OCCL_PARAMS_ALREADY_SET(0x19.toByte()),
    OCCL_PARAM(0x1a.toByte()),
    ILLEGAL_CDTHR_VALUE(0x1b.toByte()),
    IGNORE_COMMAND(0x1c.toByte()),
    INVALID_CRC(0x1d.toByte()),
    UNKNOWN(0xff.toByte());
}
